reasoner_validator/biolink/__init__.py

```diff
@@ -317,14 +300,99 @@ def validate_edge(self, edge: Dict):
 
         if self.graph_type is TrapiGraphType.Knowledge_Graph:
             if not attributes:
-                # TODO: not quite sure whether and how to fully validate the 'attributes' of an edge
                 # For now, we simply assume that *all* edges must have *some* attributes
                 # (at least, provenance related, but we don't explicitly test for them)
                 self.report_error(f"Edge '{edge_id}' has missing or empty attributes?")
+            else:
+                # TODO: attempt some deeper attribute validation here
+                for attribute in attributes:
+                    attribute_type_id: str = attribute['attribute_type_id']
+                    #
+                    # TODO: not sure if this should only be a Pytest 'warning' rather than an Pytest 'error'
+                    #
+                    if not self.is_curie(attribute_type_id):
+                        self.report_error(
+                            f"Edge '{edge_id}' attribute_type_id '{str(attribute_type_id)}' is not a CURIE?"
```

Comment:
I would say warning is probably more beneficial as attributes_types are not always well standardized.

Reply:
At the moment, all discrepancies are reported as 'error' but Eric has made the same point, and also, some normalization of errors into an index set of error codes (with parameters) is on the near term horizon, maybe over the next few days. Afterwards, it may be useful to decide which items are absolute errors, and which could be warnings or even info items.